tests/vm_test.go

```diff
@@ -1326,61 +1323,59 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					By("Waiting for VM to be ready")
 					Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
 				})
-				DescribeTable("with a failing VMI and the kubevirt.io/keep-launcher-alive-after-failure annotation", func(keepLauncher string) {
-					// The estimated execution time of one test is 400 seconds.
-					By("Creating a Kernel Boot VMI with a mismatched disk")
-					vmi := utils.GetVMIKernelBootWithRandName()
-					vmi.Spec.Domain.Firmware.KernelBoot.Container.Image = cd.ContainerDiskFor(cd.ContainerDiskCirros)
 
-					By("Creating a VM with RunStrategyManual")
-					vm := libvmi.NewVirtualMachine(vmi)
+				DescribeTable("with a failing VMI and the kubevirt.io/keep-launcher-alive-after-failure annotation", func(keepLauncher string) {
+					By("creating a Running VM")
+					vm := libvmi.NewVirtualMachine(libvmifact.NewCirros(
+						libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+						libvmi.WithNetwork(v1.DefaultPodNetwork()),
+						libvmi.WithAnnotation(v1.KeepLauncherAfterFailureAnnotation, keepLauncher),
+						libvmi.WithAnnotation(v1.FuncTestLauncherFailFastAnnotation, ""),
+					), libvmi.WithRunStrategy(v1.RunStrategyOnce))
 					vm.Spec.Running = nil
-					vm.Spec.RunStrategy = &runStrategyManual
 
-					By("Annotate the VM with regard for leaving launcher pod after qemu exit")
-					vm.Spec.Template.ObjectMeta.Annotations = map[string]string{
-						v1.KeepLauncherAfterFailureAnnotation: keepLauncher,
-					}
 					vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, k8smetav1.CreateOptions{})
 					Expect(err).ToNot(HaveOccurred())
 
-					By("Starting the VMI with virtctl")
-					startCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_START, "--namespace", vm.Namespace, vm.Name)
-					Expect(startCommand()).To(Succeed())
-
-					By("Waiting for VM to be in Starting status")
-					Eventually(ThisVM(vm), 160*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusStarting))
+					By("waiting for virt-launcher")
+					var vmi *v1.VirtualMachineInstance
+					Eventually(func() bool {
+						vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+						if errors.IsNotFound(err) {
+							return false
+						}
+						Expect(err).ToNot(HaveOccurred())
+						_, err = libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
+						if errors.IsNotFound(err) {
+							return false
+						}
+						Expect(err).ToNot(HaveOccurred())
+						return true
+					}, 60*time.Second, 5*time.Second).Should(BeTrue(), "virt-launcher should be created eventually")
```

Comment:
Please use explicit intervals:
```suggestion
					}).WithTimeout(60*time.Second).WithPolling(5*time.Second).Should(BeTrue(), "virt-launcher should be created eventually")
```

Reply:
Done.